pyproject.toml

```diff
@@ -27,9 +27,9 @@ requires = [
     "Cython>=3.0.8,<3.2.0",
     # The upper bound on pybind11 is pre-emptive only
     "pybind11>=2.13.2,<2.14.0",
-    # The upper bound on pythran is pre-emptive only; 0.17.0
+    # The upper bound on pythran is pre-emptive only; 0.18.0
     # is released/working at the time of writing
-    "pythran>=0.14.0,<0.18.0",
+    "pythran>=0.14.0,<0.19.0",
```

Comment:
See discussion at https://github.com/scipy/scipy/pull/23013#discussion_r2096347143.

With `pythran` `0.18.0` installed alongside this branch locally, the full testsuite seemed happy on x86-64 linux via `spin test -j 32`.